app/javascript/dashboard/mixins/fileUploadMixin.js

```diff
@@ -0,0 +1,72 @@
+import {
+  MAXIMUM_FILE_UPLOAD_SIZE,
+  MAXIMUM_FILE_UPLOAD_SIZE_TWILIO_SMS_CHANNEL,
+} from 'shared/constants/messages';
+import { checkFileSizeLimit } from 'shared/helpers/FileHelper';
+import { DirectUpload } from 'activestorage';
+
+export default {
+  methods: {
+    onFileUpload(file) {
+      if (this.globalConfig.directUploadsEnabled) {
+        this.onDirectFileUpload(file);
+      } else {
+        this.onIndirectFileUpload(file);
+      }
+    },
+    onDirectFileUpload(file) {
```

Comment:
Let's try to write specs for these

Reply:
@scmmishra Added here https://github.com/chatwoot/chatwoot/pull/7698/files#diff-74f691acb76614d66a0f6f5e3d9cb71a1a19b969f0fa51ffb9ead3246372ee55R1